index.d.ts

```diff
@@ -0,0 +1,88 @@
+/**
+ * Style of the box border.
+ */
+export interface BorderStyle {
+	readonly topLeft: string;
+	readonly topRight: string;
+	readonly bottomLeft: string;
+	readonly bottomRight: string;
+	readonly horizontal: string;
+	readonly vertical: string;
+}
+
+/**
+ * Spacing used for `padding` and `margin`.
+ */
+export interface Spacing {
+	readonly top: number;
+	readonly right: number;
+	readonly bottom: number;
+	readonly left: number;
+}
+
+export interface Options {
+	/**
+	 * Color of the box border.
+	 */
+	readonly borderColor?: string;
+
+	/**
+	 * Style of the box border.
+	 *
+	 * @default 'single'
+	 */
+	readonly borderStyle?: string | BorderStyle;
```

Comment:
Can you either manually define the supported styles (https://github.com/sindresorhus/cli-boxes#cliboxes) or submit a TS definition to `cli-boxes` and import the definition here?

Reply:
By manually define, do you mean like this?:

```js
type Border = 'classic' | 'double' | 'double-single' | 'round' | 'single-double' | 'single';
```

I've opened the PR, but I can do both, to make it easier for those going from JS to TS.

Reply:
```suggestion
	readonly borderStyle?: 'classic' | 'double' | 'double-single' | 'round' | 'single-double' | 'single' | BorderStyle;
```

Or is this preferred?

Reply:
Can we use a const string enum for this?